docs/REVIEWING.md

```diff
@@ -73,6 +73,17 @@ Documentation is incredibly important to Vector; it is a feature and
 differentiator for Vector. Pull requests should not be merged without adequate
 documentation, nor should they be merged with "TODOs" opened for documentation.
 
+Ideally all modules should have module level documentation. Module level
+documentation can be omitted for modules where the purpose is obvious. For
+example the with the sinks there is typically a standard number of modules
+included (service.rs, sink.rs, request_builder.rs), these modules don't need
```

Comment:
It may be useful to have a high level doc somewhere that explains the differences and function of each, as it may not be entirely obvious to newcomers. Though I agree copy-pasting the same comment for every component is wasteful.

Reply:
Big +1 on a high level doc.

Reply:
I feel like something was probably written when the new pattern was merged in originally - but I think it might have just been in the PR or RFC.

Reply:
Along those lines, we should perhaps have guidance or rules on when the standard modules aren't needed (not all new style sinks implement all the same modules).